src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -839,6 +764,8 @@ private long getOffset() {
           name ->
               pipeStatus.updateCommitLag(currentTime - FileNameUtils.fileNameToTimeIngested(name)));
 
+      ingestionService.ingestFiles(fileNamesCopy);
```

Comment:
lets put this above line 758

We want to update committedOffset only after ingestFiles API have succeeded. 

(This file is the only file which could have potential bugs, lets make sure we revert back to the same changes before exactly once change was introduced)

Reply:
moved the line.

good point, i just did a diff against the 1.6.9 tag, and excluding the log and partitionbuffer changes, the only major change is the 'shouldSkipNullValue' method.